### PR TITLE
🚀 Add workflow to clean PR caches

### PR DESCRIPTION
**Pull request by Zaid-maker**
⚠️⚠️⚠️ Since we do not accept all types of pull requests and do not want to waste your time. Please be sure that you have read pull request rules:
https://github.com/louislam/uptime-kuma/blob/master/CONTRIBUTING.md#can-i-create-a-pull-request-for-uptime-kuma

Tick the checkbox if you understand [x]: 
- [x] I have read and understand the pull request rules.

# Description

This workflow will delete all unused caches as i saw in caches it's reaching its limit of (9.82 GB of 10 GB Used)

## Type of change

Please delete any options that are not relevant.

- New feature (non-breaking change which adds functionality)
- Other

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and tested it
- [x] I have commented my code, particularly in hard-to-understand areas
  (including JSDoc for methods)
- [x] My changes generate no new warnings
- [ ] My code needed automated testing. I have added them (this is optional task)

## Screenshots (if any)

image

### CODE REVIEW

**Comment by CommanderStorm** [comment]
Why is reaching the limit bad?
The cache you mentioned is a FiFo cache ⇒ reaching a limit does not prevent further usage afaik.

**Comment by Zaid-maker** [comment]
> Why is reaching the limit bad?
> The cache you mentioned is a FiFo cache ⇒ reaching a limit does not prevent further usage afaik.

When limit reached actions will not run further and will give errors

**Comment by CommanderStorm** [comment]
See https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#usage-limits-and-eviction-policy

> GitHub will remove any cache entries that have not been accessed in over 7 days. There is no limit on the number of caches you can store, but the total size of all caches in a repository is limited to 10 GB.
> 
> If you exceed the limit, GitHub will save the new cache but will begin evicting caches until the total size is less than the repository limit. The cache eviction process may cause cache thrashing, where caches are created and deleted at a high frequency. To reduce this, you can review the caches for a repository and take corrective steps, such as removing caching from specific workflows.

=> I don't think this is a problem.. Why do you think it is?

**Comment by Zaid-maker** [comment]
> See https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#usage-limits-and-eviction-policy
> 
> > GitHub will remove any cache entries that have not been accessed in over 7 days. There is no limit on the number of caches you can store, but the total size of all caches in a repository is limited to 10 GB.
> > If you exceed the limit, GitHub will save the new cache but will begin evicting caches until the total size is less than the repository limit. The cache eviction process may cause cache thrashing, where caches are created and deleted at a high frequency. To reduce this, you can review the caches for a repository and take corrective steps, such as removing caching from specific workflows.
> 
> => I don't think this is a problem.. Why do you think it is?

Because when the pR's got merged they left a cache behind which i think will not get used anywhere, so its good to delete those caches

**Comment by CommanderStorm** [comment]
That does not answer the question:
> Why is reaching the limit bad?

I think your base assumption is not quite right. Having a full cache is not a bad thing. If a branch has been merged, the changes from this branch are now in master.
⇒ deleting them would probably cause a cache-miss

File: .github/workflows/pr-clean-caches.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+name: Cleanup branch caches
+
+on:
+  pull_request:
+    types: [opened, closed, reopened, synchronize]
+  push:
```
Comment (CommanderStorm) [inline .github/workflows/pr-clean-caches.yml:6]: This is probaly not inteded. Your PR would delete cache
- on push,
- edit of description,
- sync
- …

**Comment by Zaid-maker** [comment]
> > When limit reached actions will not run further
> 
> I don't think so, because we actually hit the limit a long time ago and I have never seen that happen.
> 
> > Least recently used caches will be automatically evicted to limit the total cache storage to 10 GB.
> 
> I think the warning message explains already, `be automatically evicted` means that the caches will be deleted automatically.

But still work flow is very helpful it's recommended by GitHub

File: .github/workflows/pr-clean-caches.yml (hunk 1 above)
Comment (Zaid-maker) [reply .github/workflows/pr-clean-caches.yml:6]: What else its doing? can't you see what types are given in it?

**Comment by Zaid-maker** [comment]
> > When limit reached actions will not run further
> 
> I don't think so, because we actually hit the limit a long time ago and I have never seen that happen.
> 
> > Least recently used caches will be automatically evicted to limit the total cache storage to 10 GB.
> 
> I think the warning message explains already, `be automatically evicted` means that the caches will be deleted automatically.

`If you exceed the limit, GitHub will save the new cache but will begin evicting caches until the total size is less than the repository limit. The cache eviction process may cause cache thrashing, where caches are created and deleted at a high frequency. To reduce this, you can review the caches for a repository and take corrective steps, such as removing caching from specific workflows`

**Comment by CommanderStorm** [comment]
If gigantic cache files are created in fast succession, this might cause thrashing.
But since we don't do that, I don't see the advantage.

**Pull request closed without merging**